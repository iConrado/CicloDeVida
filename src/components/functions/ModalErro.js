import React from 'react';

import { 
  StyleSheet, 
  View, 
  Text, 
  Modal,  
  TouchableOpacity, 
} from 'react-native';

const erroPadrao = {
  titulo: 'Erro',
  descricao: 'O aplicativo se comportou de maneira inesperada.',
  solucao: 'Tente conferir os dados, voltar uma tela ou reiniciar o app.'
};

export default class ModalErro extends React.Component {

  // Componente para apresentação de uma tela de erros
  //
  // Props esperadas: 
  // visivel  [boolean] - estado repassado para informar se a tela estará visível ou não
  // fechar() [função]  - função para ser acionada para fechar a janela 
  //                      (necessita bind no componente de origem)
  // objErro  [objeto]  - objeto contendo as descrições do erro. na ausência será retornada 
  //                      uma mensagem padrão, retornando ao aplicativo

  render() {
    const erro = this.props.objErro || erroPadrao;

    return (
      <Modal 
        transparent
        visible={this.props.visivel}
        animationType={'fade'}
        onRequestClose={() => this.props.fechar()}
      >
        <View style={styles.container}>
          <View style={styles.fundo} />
          
          <View style={styles.caixa}>
            <Text style={styles.titulo}>{erro.titulo}</Text>
            <Text style={styles.descricao}>{erro.descricao}</Text>
            <Text style={styles.solucao}>{erro.solucao}</Text>
            <TouchableOpacity
              style={styles.botao}
              onPress={() => this.props.fechar()}
            >
              <Text style={styles.textoBotao}>OK</Text>
            </TouchableOpacity>
          </View>
        </View>
      </Modal>
    );
  }
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'center',
  },
  fundo: {
    position: 'absolute',
    flex: 1,
    width: '100%',
    height: '100%',
    justifyContent: 'center',
    alignItems: 'center',
    backgroundColor: '#000',
    opacity: 0.8,
    zIndex: 1,
  },
  caixa: {
    position: 'absolute',
    width: '80%',
    justifyContent: 'center',
    backgroundColor: '#FFF',
    borderRadius: 10,
    paddingHorizontal: 5,
    paddingVertical: 10,
    zIndex: 2
  },
  titulo: {
    alignSelf: 'center',
    fontWeight: 'bold',
    fontSize: 18,
    marginBottom: 5,
  },
  descricao: {
    fontSize: 14,
    marginBottom: 5,
  },
  solucao: {
    fontSize: 14,
  }, 
  textoBotao: {
    fontSize: 16,
    fontWeight: 'bold',
    color: '#FFF',
    marginHorizontal: 5,
  },
  botao: {
    alignSelf: 'center',
    alignItems: 'center',
    minWidth: '30%',
    paddingVertical: 4,
    marginTop: 10,
    borderRadius: 2,
    elevation: 4,
    backgroundColor: '#5C86F5'
  },
});
