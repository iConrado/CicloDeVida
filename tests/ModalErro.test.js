import 'react-native';
import renderer from 'react-test-renderer';

import React from 'react';
import ModalErro from '../src/components/functions/ModalErro';

it('renders correctly', () => {
  const rendered = renderer.create(<ModalErro />).toJSON();
  expect(rendered).toMatchSnapshot();
});
