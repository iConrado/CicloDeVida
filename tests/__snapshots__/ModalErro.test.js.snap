// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  onRequestClose={[Function]}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "#000",
          "flex": 1,
          "height": "100%",
          "justifyContent": "center",
          "opacity": 0.8,
          "position": "absolute",
          "width": "100%",
          "zIndex": 1,
        }
      }
    />
    <View
      style={
        Object {
          "backgroundColor": "#FFF",
          "borderRadius": 10,
          "justifyContent": "center",
          "paddingHorizontal": 5,
          "paddingVertical": 10,
          "position": "absolute",
          "width": "80%",
          "zIndex": 2,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "alignSelf": "center",
            "fontSize": 18,
            "fontWeight": "bold",
            "marginBottom": 5,
          }
        }
      >
        Erro
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 14,
            "marginBottom": 5,
          }
        }
      >
        O aplicativo se comportou de maneira inesperada.
      </Text>
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        style={
          Object {
            "fontSize": 14,
          }
        }
      >
        Tente conferir os dados, voltar uma tela ou reiniciar o app.
      </Text>
      <View
        accessibilityComponentType={undefined}
        accessibilityLabel={undefined}
        accessibilityTraits={undefined}
        accessible={true}
        collapsable={undefined}
        hasTVPreferredFocus={undefined}
        hitSlop={undefined}
        isTVSelectable={true}
        nativeID={undefined}
        onLayout={undefined}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "alignSelf": "center",
            "backgroundColor": "#5C86F5",
            "borderRadius": 2,
            "elevation": 4,
            "marginTop": 10,
            "minWidth": "30%",
            "opacity": 1,
            "paddingVertical": 4,
          }
        }
        testID={undefined}
        tvParallaxProperties={undefined}
      >
        <Text
          accessible={true}
          allowFontScaling={true}
          ellipsizeMode="tail"
          style={
            Object {
              "color": "#FFF",
              "fontSize": 16,
              "fontWeight": "bold",
              "marginHorizontal": 5,
            }
          }
        >
          OK
        </Text>
      </View>
    </View>
  </View>
</Modal>
`;
